From 56e87f13411ca8ccecf2ffc457225ccebf44f7a3 Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Sun, 17 May 2026 20:21:29 +0300
Subject: [PATCH 1/2] Use relative library path, enable layer explicitly

---
 low_latency_layer.json.in | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/low_latency_layer.json.in b/low_latency_layer.json.in
index e93b438..fc960a2 100644
--- a/low_latency_layer.json.in
+++ b/low_latency_layer.json.in
@@ -3,7 +3,7 @@
     "layer": {
         "name": "VK_LAYER_KORTHOS_low_latency",
         "type": "GLOBAL",
-        "library_path": "@CMAKE_INSTALL_FULL_LIBDIR@/libVkLayer_KORTHOS_LowLatency.so",
+        "library_path": "libVkLayer_KORTHOS_LowLatency.so",
         "api_version": "1.3.0",
         "implementation_version": "3",
         "description": "Input latency reduction layer",
@@ -11,8 +11,11 @@
             "vkGetInstanceProcAddr": "LowLatency_GetInstanceProcAddr",
             "vkGetDeviceProcAddr": "LowLatency_GetDeviceProcAddr"
         },
+        "enable_environment": {
+            "LOW_LATENCY_LAYER": "1"
+        },
         "disable_environment": {
             "DISABLE_LOW_LATENCY_LAYER": "1"
         }
     }
-}
\ No newline at end of file
+}
-- 
2.54.0

