From 2be394500cf5e5ac4908d72d315977d32b67e3ab Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Sat, 17 Jan 2026 00:26:46 +0200
Subject: [PATCH 4/4] upscalers: also test if new files exist in the tracking
 file

---
 upscalers.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/upscalers.py b/upscalers.py
index 12122a3..f39e00f 100644
--- a/upscalers.py
+++ b/upscalers.py
@@ -187,8 +187,9 @@ def __check_upscaler_files(
         with open(version_file, encoding='utf-8') as version_fd:
             version = version_fd.read()
         version = json.loads(version)
-        # test if new attributes as exist in the config
-        _ = version[tuple(version.keys())[0]].get('md5_hash')
+        # test if new files and their attributes exist in the tracking file
+        for dst in files.keys():
+            _ = version[dst].get('md5_hash')
     except Exception as e:
         log.warn(f'Error while reading version file "{version_file}"')
         log.warn(repr(e))
-- 
2.52.0

