From 53276a866e01564cf807217619183e6a9106f04b Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Thu, 24 Apr 2025 01:02:04 +0300
Subject: [PATCH] Makefile: fix r1kab3rn-isms

---
 Makefile | 22 ++++++++++------------
 1 file changed, 10 insertions(+), 12 deletions(-)

diff --git a/Makefile b/Makefile
index aae5d9b..4c3540c 100644
--- a/Makefile
+++ b/Makefile
@@ -111,21 +111,20 @@ libmspack-install: libmspack-dist
 # unzip
 #
 
-DEB_HOST_GNU_TYPE := $(shell dpkg-architecture -qDEB_HOST_GNU_TYPE)
-CC = $(DEB_HOST_GNU_TYPE)-gcc
-CFLAGS := $(shell dpkg-buildflags --get CFLAGS)
-LDFLAGS := $(shell dpkg-buildflags --get LDFLAGS)
-CPPFLAGS := $(shell dpkg-buildflags --get CPPFLAGS)
+# Flags are from Proton
+CFLAGS ?= -O2 -march=nocona -mtune=core-avx2
+LDFLAGS ?= -Wl,-O1,--sort-common,--as-needed
 DEFINES = -DACORN_FTYPE_NFS -DWILD_STOP_AT_DIR -DLARGE_FILE_SUPPORT \
  -DUNICODE_SUPPORT -DUNICODE_WCHAR -DUTF8_MAYBE_NATIVE -DNO_LCHMOD \
  -DDATE_FORMAT=DF_YMD -DUSE_BZIP2 -DIZ_HAVE_UXUIDGID -DNOMEMCPY \
  -DNO_WORKING_ISPRINT
+UNZIP_PATCHES := $(shell cat subprojects/unzip/debian/patches/series)
 
 $(OBJDIR)/.build-unzip-dist: | $(OBJDIR)
 	$(info :: Building unzip )
 	cd subprojects/unzip && \
-	dpkg-source --before-build . && \
-	make -f unix/Makefile prefix=/usr D_USE_BZ2=-DUSE_BZIP2 L_BZ2=-lbz2 CC="$(CC) -Wall" LF2="$(LDFLAGS)" CF="$(CFLAGS) $(CPPFLAGS) -I. $(DEFINES)" unzips
+	$(foreach pch, $(UNZIP_PATCHES),patch -Np1 -i debian/patches/$(pch) &&) \
+	make -f unix/Makefile prefix=/usr D_USE_BZ2=-DUSE_BZIP2 L_BZ2=-lbz2 LF2="$(LDFLAGS)" CF="$(CFLAGS) -I. $(DEFINES)" unzips
 	touch $(@)
 
 .PHONY: unzip-dist
-- 
2.49.0

