From df6fce83929d77806dd848ddbedd80b67aba936e Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Mon, 12 May 2025 18:09:15 +0300
Subject: [PATCH] Makefile: skip python-xlib

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index a0d94bb..04b120f 100644
--- a/Makefile
+++ b/Makefile
@@ -11,12 +11,12 @@ x86_64_LIBDIR  = $(BASEDIR)/lib/x86_64-linux-gnu
 
 .PHONY: all
 
-all: cabextract-dist libmspack-dist unzip-dist python-xlib-dist
+all: cabextract-dist libmspack-dist unzip-dist
 
 .PHONY: install
 
 # Note: `export DEB_BUILD_MAINT_OPTIONS=hardening=-format` is required for the unzip target
-install: protonfixes-install cabextract-install libmspack-install unzip-install python-xlib-install
+install: protonfixes-install cabextract-install libmspack-install unzip-install
 
 #
 # protonfixes
-- 
2.49.0

